### ISSUE TITLE
FR: Add Fedora Silverblue install instructions to documentation

### ISSUE THREAD

**Issue by telenieko** [tailscale/tailscale#6761]
### What are you trying to do?

I want to install Tailscale on Fedora Silverblue

### How should we solve this?

For now, ddd instructions in the documentation, as Fedora Silverblue does not work exactly as Fedora (which is documented).

```
sudo wget https://pkgs.tailscale.com/stable/fedora/tailscale.repo -P /etc/yum.repos.d/
# now disable repo_gpgcheck & gpgcheck on /etc/yum.repos.d/tailscale.repo
sudo rpm-ostree install --apply-live tailscale
```

This would be a workaround, ideally there would be a flatpak package for tailscale.

### What is the impact of not solving this?

See #5582 and [fedora forum](https://discussion.fedoraproject.org/t/how-to-add-an-additional-repo-file-on-silverblue/35641/4) for instances of people asking for this.

Not solving this makes Tailscale harder to use on Fedora Silverblue, though not impossible

**Comment by Justinzobel** [comment]
I only had to disable gpgcheck when installing, but came across this info after. I'd love for this to be added to the documentation for Fedora Immutable (might use this name as Silverblue/Kinoite/Sericea will take up a lot of room)

**Comment by vielmetti** [comment]
Successfully brought up Tailscale on Fedora Linux 38 "Kinoite" using these instructions, after failing with the "dnf" based instructions on the web site.

**Comment by philcryer** [comment]
This script in the ublue (Universal Blue) repo works perfectly on Silverblue and Bluefin
https://github.com/castrojo/ublue/blob/main/bits/tailscale

**Comment by Justinzobel** [comment]
We're coming up on the 1-year anniversary for a documentation change proposal. Any chance we can get this merged?

**Comment by philcryer** [comment]
Looking at `scirpts/installer.sh` could we make a case for silverblue? Looking at os-release:

```
❯ cat /etc/os-release 
NAME="Fedora Linux"
VERSION="39.20231204.0 (Silverblue)"
ID=fedora
VERSION_ID=39
VERSION_CODENAME=""
PLATFORM_ID="platform:f39"
PRETTY_NAME="Fedora Linux 39.20231204.0 (Bluefin)"
ANSI_COLOR="0;38;2;60;110;180"
LOGO=fedora-logo-icon
CPE_NAME="cpe:/o:fedoraproject:fedora:39"
DEFAULT_HOSTNAME="fedora"
HOME_URL="https://silverblue.fedoraproject.org"
DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora-silverblue/"
SUPPORT_URL="https://ask.fedoraproject.org/"
BUG_REPORT_URL="https://github.com/fedora-silverblue/issue-tracker/issues"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=39
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=39
SUPPORT_END=2024-11-12
VARIANT="Silverblue"
VARIANT_ID=silverblue
OSTREE_VERSION='39.20231204.0'
```

But I'm having issues getting it to catch:

```
                        silverblue)
                                OS="$ID"
                                VERSION="$(echo "$VARIANT_ID")"
                                PACKAGETYPE="rpm-ostree"
                                ;;
```

Doesn't work, and still goes to Fedora/dnf

```
Installing Tailscale for fedora, using method dnf
+ sudo dnf install -y 'dnf-command(config-manager)'
```

Any ideas on getting that to catch? This would be a straightforward way to make it just work like all of the other Linux variants.

**Comment by chapmanjacobd** [comment]
As far as I can tell this is the current idiomatic way:

```sh
sudo ostree remote add tailscale https://pkgs.tailscale.com/stable/fedora/tailscale.repo
sudo rpm-ostree install tailscale
sudo systemctl reboot

# after reboot
sudo systemctl enable --now tailscaled
sudo tailscale up
```

**Comment by philcryer** [comment]
Thanks, this was last year when I was on Silverblue, I've since migrated to Bluefin https://projectbluefin.io/ which has Tailscale support built-in. Appreciate the followup though!

**Comment by Justinzobel** [comment]
Yay we made it to two years! :partying_face:

**Comment by jlzhjp** [comment]
@chapmanjacobd I don't think adding an rpm repo as an ostree remote makes sense here.

Tailscale is now in the official Fedora repositories (https://packages.fedoraproject.org/pkgs/tailscale/tailscale/), so you can just run:

```bash
sudo rpm-ostree install tailscale
# reboot
sudo systemctl enable --now tailscaled
sudo tailscale up
```

**Comment by xrazis** [comment]
~The Fedora repo has a slightly outdated version (1.84 vs 1.90 at the time of writing), so adding an ostree remote makes sense if you want to keep up with Tailscale's release cycle.~

See comments bellow.

**Comment by jlzhjp** [comment]
Hi, @xrazis 

Edit: There is a detailed comparison of rpm repos and ostree remotes [here](https://discussion.fedoraproject.org/t/how-to-add-an-additional-repo-file-on-silverblue/35641/11).

I don't think this is a valid command, ostree remote is more like a git remote imo, adding a rpm `.repo` makes no sense here. Does it actually work?

```
sudo ostree remote add tailscale https://pkgs.tailscale.com/stable/fedora/tailscale.repo
```

To layer the tailscale package I think OP's command is correct.

```
sudo wget https://pkgs.tailscale.com/stable/fedora/tailscale.repo -P /etc/yum.repos.d/
```

It might also be worth filing a bug or opening a PR against the Fedora package so the maintainers can update the package accordingly.

**Comment by xrazis** [comment]
Yes, you are right, adding Tailscale's .repo is what I did and meant to write.

_I confused the terminology as I am new to atomic desktops. I've edited my previous comment as to avoid confusing anyone reading this._
